pyttman/core/internals.py

```diff
@@ -60,10 +69,20 @@ def __init__(self, **kwargs):
         self.LOG_FORMAT: str | None = None
         self.LOG_TO_STDOUT: bool = False
 
-        [setattr(self, k, v) for k, v in kwargs.items()
+        [self.__set_attr(k, v) for k, v in kwargs.items()
          if not inspect.ismodule(v)
          and not inspect.isfunction(v)]
 
+    def __set_attr(self, k, v):
+        tmp = v
+        if isinstance(v, dict):
+            tmp = self.__dict2obj(v)
+            
+        setattr(self, k, tmp)
+        
+    def __dict2obj(self, dictionary):
+        return json.loads(json.dumps(dictionary), object_hook=CustomUserDict)
```

Comment:
I have some thoughts about this method. 

1. Is there a specific reason as to why `__` is used? I don't think it should be named with two leading underscores, since name mangling here is unlikely to be utilized. 
I would prefer a single `_` in the method and attribute name.

2. Is dumping the entire dictionary to a string and then reloading it again in to the custom dict class `CustomUserDict` really necessary? Can it not be done with a cast? 

3. I can't see `self` being used here. I think this should be a class method in the CustomUserDict instead, returning an instance of itself.